docs/sec-blueprint/07-Hardening.md

```diff
@@ -0,0 +1,597 @@
+
+  Disable the process\_vm\_\*v syscalls which allow one process to
+  peek/poke the virtual memory of another. This configuration is
+  supported in Linux 3.5 and greater and thus should only be disabled
+  for such versions.
+```
+  CROSS_MEMORY_ATTACH=n
+```
+
+### Disable core dumps
+
+  Core dumps provide lot of debug information for hackers. So disabling
+  core dumps is recommended in production builds. This configuration is
+  supported in Linux 3.7 and greater and thus should only be disabled
+  for such versions.
+
+```
+  CONFIG_COREDUMP=n
+```
+
+### Disable Legacy Linux Support
+
+  There are some Kernel Configs which are present only to support legacy
+  binaries. See also section 2.2.2.18 for disabling support for legacy
+  binary formats. The uselib system call, in particular, has no valid
+  use in any libc6 or uclibc system in recent times. This configuration
+  is supported in Linux 3.15 and greater and thus should only be
+  disabled for such versions.
+
+```
+  CONFIG_USELIB=n
+```
+
+### Disable firmware auto-loading user mode helper
+
+  The firmware auto loading helper, which is a utility executed by the
+  kernel on hotplug events requiring firmware, needs to be set setuid.
+  As a result of this, the helper utility is an attractive target for
+  attackers with control of physical ports on the device. Disabling this
+  configuration is supported in Linux 3.9 and greater.
+
+```
+  CONFIG_FW_LOADER_USER_HELPER=n
+```
+
+### Enable Kernel Panic on OOPS
+
+  When fuzzing the kernel or attempting kernel exploits attackers are
+  likely to trigger kernel OOPSes. Setting the behavior on OOPS to PANIC
+  can impede their progress. This configuration is supported in Linux
+  3.5 and greater and thus should only be enabled for such versions.
+
+```
+  CONFIG_PANIC_ON_OOPS=y
+```
+
+### Disable socket monitoring interface
+
+  These monitors can be used to inspect shared file descriptors on Unix
+  Domain sockets or traffic on ‘localhost’ which is otherwise assumed to
+  be confidential. The **CONFIG\_PACKET\_DIAG** configuration is
+  supported in Linux 3.7 and greater and thus should only be disabled
+  for such versions. The **CONFIG\_UNIX\_DIAG** configuration is
+  supported in Linux 3.3 and greater and thus should only be disabled
+  for such versions.
+
+```
+  CONFIG_PACKET_DIAG=n
+  CONFIG_UNIX_DIAG=n
+```
+
+### Disable BPF JIT
+
+  The BPF JIT can be used to create kernel-payloads from firewall table
+  rules. This configuration for the MIPS architecture is supported in
+  Linux 3.16 and greater and thus should only be disabled for such
+  versions.
+```
+  BPF_JIT=n
+```
+
+### Disable checkpoint/restore
+
+  The checkpoint/restore service can take a process, freeze it and
+  migrate it. This results in providing more info than a core dump. This
+  configuration is supported in Linux 3.3 and greater and thus should
+  only be disabled for such versions.
+
+```
+  CONFIG_CHECKPOINT_RESTORE=n
```

Comment:
!! This is actually meant as feature !! 
We have to investigate how this can work with the app lifetime mode.